aptos-move/block-executor/src/scheduler.rs

```diff
@@ -826,4 +838,17 @@ impl Scheduler {
     fn done(&self) -> bool {
         self.done_marker.load(Ordering::Acquire)
     }
+
+    fn txn_dependency_by_index(&self, index: TxnIndex) -> &CachePadded<Mutex<Vec<TxnIndex>>> {
+        let pos = self.index_mapping.positions_by_index[index as usize];
```

Comment:
Use position_by_index API instead of the field directly. Also, make `positions_by_index` private.

Reply:
fixed